api/v1alpha1/runner_types.go

```diff
@@ -94,6 +94,9 @@ type RunnerSpec struct {
 	DockerEnabled *bool `json:"dockerEnabled,omitempty"`
 	// +optional
 	DockerMTU *int64 `json:"dockerMTU,omitempty"`
+
+	// +optional
+	RunnerVolumeSizeLimit string `json:"runnerVolumeSizeLimit,omitempty"`
```

Comment:
Anything related to the runner container in the runner pod is currently not prefixed(like `Image`, `Env`, `EnvFrom`) so just for consistency, you could just use `VolumeSizeLimit`.

Also, I thought controller-runtime(although it's often said to be a part of kubebuilder) had support for Quantity field so you could directly use `*resource.Quantity` instead of `string`.

```suggestion
	VolumeSizeLimit *resource.Quantity `json:"volumeSizeLimit,omitempty"`
```